### DISCUSSION TITLE
Integration with Neo4j

### DISCUSSION THREAD

**Discussion by imjaleel** [AbsaOSS/spline#1198] [Q&A]
Hello, 

We've an architecture that uses neo4j as a GraphDB, is there a way we can use Neo4j(Hosted on an EC2 Instance) as the storage DB instead of ArangoDB?

**Comment by cerveada** [comment]
The Spline server is tightly integrated with ArangoDb, so there is no simple way how to swap the db. It would mean replacing the whole server, or at least significant part of it. The Agent and UI are not dependent on the db at all, so they are independent, only relying on the Rest APIs.

**Reply by imjaleel** [reply]
Understood. Thanks for that. 

Is there any documentation that explains how we can interpret the json output of the lineage from the Spark Agent?  This way maybe I can work on a package that reads this json and build a graph in Neo4J or other similar databases.

**Reply by cerveada** [reply]
If you run the Spline server and access it via web browser, it provides documentation for both: Producer (agent facing) API and Consumer (UI facing) API.
